cmake/dependencies.cmake

```diff
@@ -0,0 +1,4 @@
+find_package(fmt REQUIRED)
+find_package(glfw3 REQUIRED)
+find_package(GTest REQUIRED)
```

Comment:
Some libraries includes the necessary dependencies for testing only when testing is needed. Might be better in the `unit-test`'s cmake file.

Reply:
Done.